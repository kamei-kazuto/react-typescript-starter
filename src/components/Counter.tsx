import * as React from 'react'
import { counterStore } from '@/store'
export interface Props {
  count: number
  increment?: () => void
  decrement?: () => void
  reset?: () => void
}

export const Component: React.SFC<Props> = ({
  count,
  increment,
  decrement,
  reset
}) => {
  const hancleClick = (action: () => void) => (
    e: React.MouseEvent<HTMLButtonElement>
  ) => action()
  return (
    <div>
      <h3>COUNTER</h3>
      <div>{count}</div>
      <button onClick={hancleClick(increment)}>+1</button>
      <button onClick={hancleClick(decrement)}>-1</button>
      <button onClick={hancleClick(reset)}>Reset</button>
    </div>
  )
}

export const Counter: React.SFC<{}> = () => {
  const {
    state: { count },
    increment,
    decrement,
    reset
  } = counterStore()

  return (
    <Component
      count={count}
      increment={increment}
      decrement={decrement}
      reset={reset}
    />
  )
}
