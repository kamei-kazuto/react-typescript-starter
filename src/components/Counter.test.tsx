import * as React from 'react'
import { mount } from 'enzyme'
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import { Component as Counter } from './Counter'

describe('Counter コンポーネント', () => {
  it('render', () => {
    const tree = mount(<Counter count={0} />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
