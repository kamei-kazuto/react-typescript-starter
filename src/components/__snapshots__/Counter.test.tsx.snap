// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter コンポーネント render 1`] = `
<Component
  count={0}
>
  <div>
    <h3>
      COUNTER
    </h3>
    <div>
      0
    </div>
    <button
      onClick={[Function]}
    >
      +1
    </button>
    <button
      onClick={[Function]}
    >
      -1
    </button>
    <button
      onClick={[Function]}
    >
      Reset
    </button>
  </div>
</Component>
`;
